### ISSUE TITLE
azurefs filling master log with errors

### ISSUE THREAD

**Issue by kveroneau** [saltstack/salt#55727]
### Description of Issue
When using the azurefs state backend the log file is filled with the following exception:
```
OSError: [Errno 2] No such file or directory: '/var/cache/salt/master/azurefs/hashes'
2019-12-22 05:14:37,392 [root             :272 ][ERROR   ][24987] Problem occurred trying to invalidate hash cach for azurefs
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/fileserver/azurefs.py", line 270, in update
    shutil.rmtree(hash_cachedir)
  File "/usr/lib/python2.7/shutil.py", line 239, in rmtree
    onerror(os.listdir, path, sys.exc_info())
  File "/usr/lib/python2.7/shutil.py", line 237, in rmtree
    names = os.listdir(path)
```

I attempted to manually create the directory mentioned, however azurefs seems to just delete it, and continues to throw the exception.  azurefs seems to work otherwise just fine.

### Setup
Configure your salt state modules to use azurefs as a root,

```
fileserver_backend:
  - gitfs
  - azurefs

azurefs:
  - account_name: Retracted
    account_key: Retracted
    container_name: Retracted
```
### Steps to Reproduce Issue
Use the azurefs fileserver_backend and watch the /var/log/salt/master log file for the exceptions.

### Versions Report

Salt Version:
           Salt: 2019.2.2
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: 1.9.0
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: 0.24.5
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.24.2
         Python: 2.7.13 (default, Sep 26 2018, 18:42:22)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1
 
System Versions:
           dist: debian 9.10 
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-11-amd64
         system: Linux
        version: debian 9.10

**Comment by garethgreenaway** [comment]
@kveroneau Thanks for the report.  I wonder if azurefs is deleting the directory but some step that would create it isn't being reached.  We'll get this one fixed.  Thanks!

**Comment by kveroneau** [comment]
Today our company noticed another potential issue with azurefs, which might be related to this issue.  AzureFS fileserver backend does not appear to be correctly hashing, as we incurred 255GB of bandwidth from our blob storage hosting our SaltStack provisioning assets.  Our assets are limited in both number and size, so 255GB of bandwidth is rather extreme.  It seems that there might be a bug in the AzureFS or Azure Python SDK in relations to hashing, as SaltStack appears to be downloading the assets from the Azure Blob storage each time it checks it.  For anybody else using AzureFS, be warned that it may incur charges you were not expecting.

We have since placed all our assets on the Salt Master filesystem and deleted the Azure blob storage.
